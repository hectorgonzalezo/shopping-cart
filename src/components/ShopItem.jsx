import React from 'react';
import PropTypes from 'prop-types';

function ShopItem({ name, img }) {
  return (
    <div className="item">
      <h1>{name}</h1>
      <img src={require(`../assets/products/${img}`)} />
    </div>
  );
}

ShopItem.defaultProps = {
  img: 'teensylc.jpeg',
};

ShopItem.propTypes = {
  name: PropTypes.string.isRequired,
  img: PropTypes.string,
};

export default ShopItem;