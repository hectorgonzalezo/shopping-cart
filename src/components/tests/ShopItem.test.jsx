import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import ShopItem from '../ShopItem';

describe('Shop items', () => {
  it('Renders correctly', () => {
    render(<ShopItem name="Car" />);

    expect(screen.getByRole('heading')).toHaveTextContent('Car');
    expect(screen.getByRole('img').getAttribute('src')).toBe('teensylc.jpeg');

    expect(screen.container).toMatchSnapshot();
  });
});
