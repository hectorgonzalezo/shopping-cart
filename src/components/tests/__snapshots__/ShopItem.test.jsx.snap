// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop items Renders correctly 1`] = `undefined`;
